source/common/upstream/upstream_impl.cc

```diff
@@ -1191,8 +1196,18 @@ void StrictDnsClusterImpl::ResolveTarget::startResolve() {
             return host->priority() == locality_lb_endpoint_.priority();
           }));
           parent_.updateAllHosts(hosts_added, hosts_removed, locality_lb_endpoint_.priority());
+        } else {
+          parent_.info_->stats().update_no_rebuild_.inc();
         }
 
+        // This makes sure parent_.all_hosts_ is updated with all resolved hosts from all
+        // priorities. This reconciliation is required since we check each new host againsts this
```

Comment:
typo "againsts"

I think this issue is confusing enough that we should have more explanation. Can you be really explicit about the flow here and why this is necessary? I think I understand but I want to make sure the next person that comes along has a really clear understanding of the bug and how we are fixing it for now. My limited understanding is that previously we would set "all hosts" to just this resolve group, which would leave to logic in the future continuing to add hosts that already exist? That's about where my understanding ends without paging back in all this code and the recent changes. So I think more explanation would be great. :)

I would potentially also reference the opened tracking issue to clean this up in the comment.

Reply:
The problem was that whenever a target resolved, it would set its hosts as `all_hosts_`, so that when another target resolved it wouldn't see its own hosts in `all_hosts_`. It would think that they're new hosts, so it would add them to its host list over and over again.

I'd probably add a comment explaining this exact thing, I think it would provide a decent explanation as to why we're doing this.

Reply:
@mattklein123 @snowp I have updated the comment accordingly.